### PR TITLE
Add test to ensure `istio-cni` disruption does not impact established ambient pod dataplane

### PR DESCRIPTION
**Pull request by bleggett**
This does nothing but add a test to ensure that disrupting `istio-cni` itself does not negatively impact established pod dataplanes.

This is not a recommendation that people SHOULD do this, as in the real world it is unsafe and would miss new pods, but this is a decent way to verify that 

- properly established pod ambient dataplanes do not have a hard, continuous dependency on `istio-cni` being healthy.
- `ztunnel` can continue to manage established pods happily even if `istio-cni` drops out

### CODE REVIEW

File: tests/integration/ambient/cni/main_test.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+		Setup(istio.Setup(&i, func(ctx resource.Context, cfg *istio.Config) {
+			// can't deploy VMs without eastwest gateway
+			ctx.Settings().SkipVMs()
+			cfg.EnableCNI = true
+			cfg.DeployEastWestGW = false
+			cfg.ControlPlaneValues = `
+values:
+  cni:
+    repair:
+      enabled: true
```
Comment (kfaseela) [inline tests/integration/ambient/cni/main_test.go:92]: Do we need a separate package for this or reuse existing cni repair folder.. this seems like using the same config or am I missing something?

Comment (bleggett) [reply tests/integration/ambient/cni/main_test.go:92]: It is, and I considered merging them, but we are deleting the daemonset (technically a destructive operation to the Istio install), so it seemed safer to use a separate Istio install entirely.

Comment (bleggett) [reply tests/integration/ambient/cni/main_test.go:92]: Lemme try merging tho.

Comment (bleggett) [reply tests/integration/ambient/cni/main_test.go]: Aight - I wanna keep it separate for now - we are basically intentionally breaking the istio install here to test behavior, and I don't want that to leak to some other unrelated test.

I removed the `repair: enabled`, that was leftover from copying the repair bits, yea. We don't want `repair` enabled for tests that aren't explicitly testing it, and there's no sidecars in this test anyway.

**Review by kfaseela — APPROVED** [review]

**Pull request merged**